src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,529 @@
+
+namespace is_liouville
+
+lemma not_liouville_zero : ¬ is_liouville 0 :=
+begin
+  classical,
+  intro h,
+  rcases h 1 with ⟨a, b, h_b_gt_1, h_abs_sub_pos, h_abs_sub_small⟩,
+  simp only [one_div, zero_sub, abs_neg, pow_one] at h_abs_sub_pos h_abs_sub_small,
+  set a' := abs a,
+  have h_a'_ineq : 0 ≤ a' := abs_nonneg _,
+  have : abs (a' / b : ℝ) = abs (a / b), { simp only [int.cast_abs, abs_div, abs_abs] },
+  rw ← this at h_abs_sub_pos h_abs_sub_small,
+  rw [abs_of_nonneg, inv_eq_one_div, div_lt_div_iff, one_mul, mul_lt_iff_lt_one_left]
+    at h_abs_sub_small,
+  { norm_cast at h_abs_sub_small,
+    rw [(by linarith : a' = 0), int.cast_zero, zero_div, abs_zero] at h_abs_sub_pos,
+    revert h_abs_sub_pos, simp only [forall_prop_of_false, not_lt] },
+  repeat { apply div_nonneg <|> norm_cast <|> linarith <|> assumption }
+end
+
+lemma abs_eval_rat_ge_one_div_denom_pow_n (f : polynomial ℤ) (a b : ℤ)
+  (h_b_pos : 0 < b) (h_not_root : (f.map (algebra_map ℤ ℝ)).eval (a / b) ≠ 0) :
+  1 / (b : ℝ) ^ f.nat_degree ≤ abs ((f.map (algebra_map ℤ ℝ)).eval (a / b)) :=
+begin
+  have h_b0 : (b : ℝ) ≠ 0, { exact_mod_cast ne_of_gt h_b_pos },
+  have h_pos : 0 < 1 / (b : ℝ) ^ f.nat_degree,
+  { exact div_pos zero_lt_one (pow_pos (by exact_mod_cast h_b_pos) _) },
+  have h_eq₁ : ((f.map (algebra_map ℤ ℝ)).eval (a / b)) =
+    1 / b ^ f.nat_degree * (∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)),
+  { -- we give a rough outline of the calculation, and prove two obligations below
+    calc  ((f.map (algebra_map ℤ ℝ)).eval (a / b))
+        = ∑ i in f.support, f.coeff i * (a / b) ^ i : _ -- (1)
+    ... = ∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i) / b ^ f.nat_degree :
+          sum_congr rfl _ -- (2)
+    ... = (∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)) / b ^ f.nat_degree :
+            by rw [sum_div]
+    ... = 1 / b ^ f.nat_degree * (∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)) :
+            by ring,
+    { -- proof of (1)
+      conv_lhs { rw [eval_map, as_sum_support f, eval₂_finset_sum] },
+      congr, funext, simp only [ring_hom.eq_int_cast, eval₂_monomial] },
+    { -- proof of (2)
+      intros i hi,
+      suffices : (b ^ f.nat_degree : ℝ) = b ^ (f.nat_degree - i) * b ^ i,
+      { field_simp [h_b0, this, mul_assoc] },
+      rw [← pow_add, nat.sub_add_cancel (le_nat_degree_of_mem_support hi)] }},
+  have h_cast_eq : ∑ i in f.support, (f.coeff i : ℝ) * a ^ i * b ^ (f.nat_degree - i) =
+                 ↑(∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)),
+  { simp only [← int.coe_cast_ring_hom, ring_hom.map_sum, ring_hom.map_mul, ring_hom.map_pow] },
+  rw [h_eq₁, h_cast_eq, abs_mul, abs_of_pos h_pos, le_mul_iff_one_le_right h_pos],
+  rw [h_eq₁, mul_ne_zero_iff, h_cast_eq] at h_not_root,
+  norm_cast at ⊢ h_not_root,
+  exact abs_pos.mpr h_not_root.2
+end
+
+lemma exists_forall_ge_of_polynomial_deriv (α : ℝ) (f : polynomial ℝ)
```

Comment:
This lemma could move to a more general file. Maybe something like `polynomial/real` or `analysis/polynomial`. I haven't thought hard about the exact location.
Also, does it help to use `Icc` in the statement, instead of `abs`? (Does it make the lemma easier to use? Does it make the lemma easier to prove?)

Reply:
- I moved this and others you'v pointed out to `analysis/real/polynomial` imitating `analysis/complex/polynomial`. 
- I've decided not to change to `Icc`, for in the proof `abs` is easier to prove.